src/wallet/rpcwallet.cpp

```diff
@@ -157,6 +158,14 @@ UniValue getnewaddress(const JSONRPCRequest& request)
     if (!request.params[0].isNull())
         strAccount = AccountFromValue(request.params[0]);
 
+    OutputType output_type = g_address_type;
+    if (!request.params[1].isNull()) {
+        output_type = ParseOutputType(request.params[1].get_str());
```

Comment:
I believe this allows the user to put in "default", which then resolves to "P2SH" no matter what the actual default `g_address_type` is set to .